public/app/features/alerting/unified/api/buildInfo.ts

```diff
@@ -57,8 +57,31 @@ export async function fetchDataSourceBuildInfo(dsSettings: { url: string; name:
     };
   }
 
-  // if no features are reported but buildinfo was return we're talking to Prometheus
-  const { features } = response.data.data;
+  // if the response looks like "data: { revision: ... }", we're talking to Loki.
+  let { revision, features, data } = response.data as any;
```

Comment:
Another way we could do this is to have a separate type from PromBuildInfoResponse for Loki responses and use a union type to see which one we get back and switch on that.

Can our code handle something like:
```ts
getBackendSrv().fetch<PromBuildInfoResponse | LokiBuildInfoResponse>
```

Reply:
This should work, you can pass a union type in to any generic and use a [user-defined type guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) to assert one or the other based on the response structure.